src/server/middleware/helmet.ts

```diff
@@ -77,15 +79,20 @@ export const helmet = koaHelmet({
       defaultSrc,
       scriptSrc: [
         "'unsafe-eval'",
-        "'unsafe-inline'",
         'https://browser.sentry-cdn.com',
         ...defaultSrc,
+        (_request: any, response: any) => {
+          response.cspNonce = uuidV4()
+          return `'nonce-${response.cspNonce}'`
+        },
       ],
       styleSrc: [
         "'unsafe-inline'",
         "'self'",
         'checkoutshopper-live.adyen.com',
         'https://tagmanager.google.com',
+        'https://www.googletagmanager.com',
+        'www.googletagmanager.com',
```

Comment:
It's not so straight forward 😄 Google lists what you need to whitelist for the GTM preview mode to work properly in the docs: https://developers.google.com/tag-manager/web/csp#preview_mode However, they actually use `https://www.googletagmanager.com` as well so that's why I added both :)

Reply:
This is such a mystery to me 😄